src/Compute/Compute/VirtualMachine/Operation/GetAzureVMBootDiagnosticsDataCommand.cs

```diff
@@ -133,19 +138,26 @@ public override void ExecuteCmdlet()
                 if (this.Windows.IsPresent
                     || (this.Linux.IsPresent && !string.IsNullOrEmpty(this.LocalPath)))
                 {
-                    var screenshotUri = new Uri(result.Body.InstanceView.BootDiagnostics.ConsoleScreenshotBlobUri);
-                    var localFile = this.LocalPath + screenshotUri.Segments[2];
-                    DownloadFromBlobUri(screenshotUri, localFile);
+                    var bootDiagnostics = this.VirtualMachineClient.RetrieveBootDiagnosticsData(this.ResourceGroupName, this.Name);
+
+                    BlobClient blobClient = new BlobClient(new Uri(bootDiagnostics.ConsoleScreenshotBlobUri));
+
+                    var localPathTest = this.LocalPath;
+                    var bootSegment = new Uri(bootDiagnostics.ConsoleScreenshotBlobUri).Segments[2];
+                    var localFile = this.LocalPath + bootSegment;
+                    blobClient.DownloadTo(localFile);
                 }
 
 
                 if (this.Linux.IsPresent)
                 {
-                    var logUri = new Uri(result.Body.InstanceView.BootDiagnostics.SerialConsoleLogBlobUri);
+                    var bootDiagnostics = this.VirtualMachineClient.RetrieveBootDiagnosticsData(this.ResourceGroupName, this.Name);
+                    var logSegment = new Uri(bootDiagnostics.SerialConsoleLogBlobUri).Segments[2];
 
-                    var localFile = (this.LocalPath ?? Path.GetTempPath()) + logUri.Segments[2];
+                    var localFile = (this.LocalPath ?? Path.GetTempPath()) + logSegment;
 
-                    DownloadFromBlobUri(logUri, localFile);
+                    BlobClient blobClient = new BlobClient(new Uri(bootDiagnostics.SerialConsoleLogBlobUri));
+                    blobClient.DownloadTo(localFile);
```

Comment:
Could you share how the Blob Uri looks  like?
And how big the blob normally is?

Reply:
Changed to using CloudBlob() instead of BlobClient.